google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataSettings.java

```diff
@@ -77,7 +74,7 @@ public final class BigtableDataSettings {
 
   private static final Logger LOGGER = Logger.getLogger(BigtableDataSettings.class.getName());
   private static final String BIGTABLE_EMULATOR_HOST_ENV_VAR = "BIGTABLE_EMULATOR_HOST";
-  private static final AtomicBoolean BUILTIN_METRICS_REGISTERED = new AtomicBoolean(false);
+  @Nullable private static Credentials credentials;
```

Comment:
The variable name is very generic and implies very generic usage. However this has a very specific usecase that is quite unepxected and a bit hacky. So this needs a better variable name something like `legacyMetricCredentialOverride` and needs a comment explaining that this to supported a deprecated method. And finally you want to mark it as `@Deprecated` to align with its purpose :)